### PR TITLE
Cylance context paths bug

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36643

## Description
The context outputs under `File` had 2 errors:
1. `Sha256` instead of `SHA256` - fix: the capitalised version was added under `File`
2.` DbotScore.Indicator` had the file's name instead of the hash - fix: the hash is now under `DbotScore.Indicator` (and the file name was removed)

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: only in cases a customer used `DbotScore.Indicator` for the file's name (which is not the usecase for that field).
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py [hunk 1]
```diff
@@ -1347,6 +1352,47 @@ def fetch_incidents():
     demisto.setLastRun({'time': current_run.isoformat().split('.')[0]})
 
 
+def add_capitalized_hash(threats_context):
```
Comment (reutshal) [inline Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py:1354]: Please improve the name of this method, it is not very self-explanatory.
I would suggest something like add_capitalized_hash_to_context.

Also, I think you can handle both cases in one method, see my suggestion

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py [hunk 2]
```diff
@@ -1347,6 +1352,47 @@ def fetch_incidents():
...
+
+    Returns:
+        threats_context(dict): dicts of context outputs for the threat of interest, containing
+        the key and value 'Sha256' (and possibly Md5) as well as the key and value 'SHA256' (and possible MD5).
+    """
+    if threat_context.get('Sha256'):
+        threat_context['SHA256'] = threat_context.get('Sha256')
+    if threat_context.get('Md5'):
+        threat_context['MD5'] = threat_context.get('Md5')
+    return threat_context
```
Comment (reutshal) [inline Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py:1392]:
```suggestion
    if not isinstance(threats_context, list):
        threats_context = [threats_context]
       
    for context_item in threats_context:
        if context_item.get('Sha256'):
            context_item['SHA256'] = context_item.get('Sha256')
        if context_item.get('Md5'):
            context_item['MD5'] = context_item.get('Md5')

    return threats_context
```

Make sure to handle the case where the context is empty / None if possible

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml [hunk 3]
```diff
@@ -265,6 +265,9 @@ script:
     - contextPath: File.SHA256
       description: The SHA256 hash of the threat.
       type: string
+    - contextPath: File.MD5
+      description: The SHA256 hash of the threat.
+      type: string
```
Comment (reutshal) [inline Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml:270]: Shouldn't new context paths need to be added to all commandsd using the new method? Is it already there?

**Review by reutshal — CHANGES_REQUESTED** [review]

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml (hunk 3 above)
Comment (Noy-Maimon) [reply Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml:270]: They are already there. The thing is that there are no File.Sha256 for any of the commands (the CamleCase version). I wasnt sure if if should add them, what do you think?

Comment (Noy-Maimon) [reply Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml:270]: They are already there. The thing is that there are no File.Sha256 for any of the commands (the CamleCase version). I wasnt sure if if should add them, what do you think?

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py (hunk 1 above)
Comment (Noy-Maimon) [reply Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py:1354]: Done

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml (hunk 3 above)
Comment (reutshal) [reply Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml:270]: Add them, if the output paths don't match the real context then it's another bug that should be solved.

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py [hunk 4]
```diff
@@ -1347,6 +1351,30 @@ def fetch_incidents():
...
+
+    Args:
+        threats_context(list): list of dicts of context outputs for the threats of interest, each containing
+        the key 'Sha256' (and possibly (Md5)).
+
+    Returns:
+        threats_context(list): list of dicts of context outputs for the threats of interest, each containing
+        the key and value 'Sha256' (and possibly Md5) as well as the key and value 'SHA256' (and possible MD5).
+    """
+    if not isinstance(threats_context, list):
```
Comment (reutshal) [inline Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py:1366]: Can there be a situation where the context is None or an empty dict? I would add a check for that

**Review by reutshal — APPROVED** [review]

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py (hunk 1 above)
Comment (Noy-Maimon) [reply Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py]: Done

**Review by reutshal — CHANGES_REQUESTED** [review]
Waiting on the yml outputs

**Review by reutshal — APPROVED** [review]

**Pull request merged**